tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_round.h

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "sfpi.h"
+
+using namespace sfpi;
+
+namespace ckernel::sfpu {
+
+inline vInt float_to_int31(vFloat v) {
+    vInt q = float_to_int16(v * 0x1p-15f, 0);
+    vInt r = float_to_int16(v - int32_to_float(q, 0) * 0x1p15f, 0);
+    return (q << 15) + r;
+}
+
+inline vFloat round_even(vFloat v) {
+    v_if (sfpi::abs(v) < 0x1p30f) {
+        v = int32_to_float(float_to_int31(v), 0);
+    }
+    v_endif;
+    return v;
+}
+
+static float pow_unsigned(float x, uint n) {
+    if (n == 0)
+        return 1.0f;
+
+    if ((n & 1) == 0)
+        return pow_unsigned(x * x, n >> 1);
+
+    return x * pow_unsigned(x * x, n >> 1);
+}
+
+template <bool APPROXIMATE, int ITERATIONS = 8>
+void calculate_round(const int decimals) {
+    const auto exp10i = [](int n) {
+        if (n < 0)
+            return pow_unsigned(0.1f, -n);
+        return pow_unsigned(10.0f, 0);
```

Comment:
Wouldn't this make any call to exp10i(x) return 1.0f if x >= 0? If that is the desired effect, can you explain a bit why that's needed in the kernel?

Reply:
Is this what was intended?

```
const auto exp10i = [](int n) {
    if (n < 0)
        return pow_unsigned(0.1f, -n);
    return pow_unsigned(10.0f, n);
};

```

Reply:
Thanks!  That was intended.  However, I found out that it's unsupported.  Zero worked because of constant folding.  I'm trying another way.

Reply:
I replaced this piece of code with LUT at 32e76c93502b293f56b31fb4e0b7f57537004f18.  It's tested with various decimal places at tenstorrent/pytorch2.0_ttnn@a5c1d24b4c4ea7e7be56f8e9c186be1d0fe2deed.

Reply:
Can you clarify what exactly was unsupported? Also, can these same kernel changes be made for Blackhole as well?

Reply:
I cannot spot exactly what was unsupported.  When low-level compilation fails, `pytest` simply hangs without printing any error message.  The best we can do is trial and error.  For example, `pow_unsigned(10.0f, 0)` succeeded and `pow_unsigned(10.0f, n)` failed.  I could infer that this operation was actually unsupported.  The former passed because constant folding optimized out the function call.